sdk/compiler/damlc/daml-stdlib-src/DA/Crypto/Text.daml

```diff
@@ -78,14 +87,106 @@ class HasToHex a where
   -- | Converts a typed data value into a hex encoded string.
   toHex: a -> BytesHex
 
+instance HasToHex Text where
+  toHex = primitive @"BEEncodeHex"
+
+-- | HIDE
+intToHexByte: Int -> BytesHex
+intToHexByte n | 0 <= n && n <= 9 = show n
+intToHexByte 10 = "a"
+intToHexByte 11 = "b"
+intToHexByte 12 = "c"
+intToHexByte 13 = "d"
+intToHexByte 14 = "e"
+intToHexByte 15 = "f"
+intToHexByte n | n < 256 = (intToHexByte(n / 16)) <> (intToHexByte(n % 16))
+intToHexByte n = error $ "Int should be in range [0, 256): " <> show n
+
+instance HasToHex Int where
+  toHex n | 0 <= n && n <= 9 = "0" <> (show n)
+  toHex 10 = "0a"
+  toHex 11 = "0b"
+  toHex 12 = "0c"
+  toHex 13 = "0d"
+  toHex 14 = "0e"
+  toHex 15 = "0f"
+  toHex n | n < 256 = intToHexByte n
+  toHex n = (toHex(n / 256)) <> (toHex(n % 256))
+
+instance HasToHex Party where
+  toHex p = toHex(partyToText p)
+
+instance NumericScale n => HasToHex (Numeric n) where
+  toHex num = toHex(show num)
```

Comment:
Isn't this a rather surprising instance? I'd suggest not offering that one. It seems very confusing to hexEncode the utf-8 encoding of the decimal string representation.

Reply:
Agreed it does look odd.

However, the numeric is (essentially) a string represented decimal (with use sites knowing what scale it should be interpreted as).

By exploiting this decimal string representation, we simplify the corresponding `fromHex` instance be an application of `parseNumeric`.

Perhaps a comment here would help to better understand this rationale?

Reply:
The rule I tend to use is to not define instances that are surprising / for which there are multiple options. For these, I instead define aptly named functions. For example,
```
-- | Convert a decimal to a hex-bytes representation by rendering it as a decimal string
-- and encoding that one using UTF-8.
decimalViaStringToHex : Decimal -> Hex
decimalVaiStringFromHex : Hex -> Optional Decimal
```